edx_rest_framework_extensions/auth/jwt/tests/test_authentication.py

```diff
@@ -534,6 +537,117 @@ def test_authenticate_jwt_and_no_session_and_set_request_user(self, mock_set_cus
         assert 'jwt_auth_mismatch_jwt_cookie_username' not in set_custom_attribute_keys
         assert response.status_code == 200
 
+    @override_settings(
+        MIDDLEWARE=(
+            'django.contrib.sessions.middleware.SessionMiddleware',
+            'django.contrib.auth.middleware.AuthenticationMiddleware',
+            'edx_rest_framework_extensions.auth.jwt.middleware.JwtAuthCookieMiddleware',
+        ),
+        ROOT_URLCONF='edx_rest_framework_extensions.auth.jwt.tests.test_authentication',
+    )
+    def test_authenticate_user_lms_and_jwt_email_mismatch_toggle_disabled(self):
+        """
+        Test success for JwtAuthentication when ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH is disabled.
+        """
+        user = factories.UserFactory(email='old@example.com')
+        jwt_header_payload, jwt_signature = self._get_test_jwt_token_payload_and_signature(user=user)
+
+        # Cookie parts will be recombined by JwtAuthCookieMiddleware
+        self.client.cookies = SimpleCookie({
+            jwt_cookie_header_payload_name(): jwt_header_payload,
+            jwt_cookie_signature_name(): jwt_signature,
+        })
+
+        # simulating email change
+        user.email = 'new@example.com'
+        user.save()  # pylint: disable=no-member
+
+        self.client.force_login(user)
+
+        response = self.client.get(reverse('authenticated-view'))
+
+        assert response.status_code == 200
+
+    @override_settings(
```

Comment:
Note: I just read about `modify_settings`. I'm not sure if you can use both, but for `EDX_DRF_EXTENSIONS`, you might be able to make the test more clear by just updating `ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH`?

Reply:
Unfortunately, `modify_settings` doesn't seem to work with the nested settings in our case. 

> but for EDX_DRF_EXTENSIONS, you might be able to make the test more clear by just updating `ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH`?

if we don't override `JWT_PAYLOAD_USER_ATTRIBUTE_MAPPING: {},` then it gets these [default values](https://github.com/openedx/edx-drf-extensions/blob/master/edx_rest_framework_extensions/settings.py#L29-L32) containing `email`. and on mismatch between the JWT and LMS user email, it simply gets updated at this part [here](https://github.com/openedx/edx-drf-extensions/blob/master/edx_rest_framework_extensions/auth/jwt/authentication.py#L206-L209). As a result, the emails get matched in our condition in `_is_jwt_and_lms_user_email_mismatch` function thus the `JwtAuthentication` returns 200 and does not fail. 

As LMS [uses](https://github.com/edx/edx-internal/blob/1301473ad43e1147d8a046056fec02e533abdb67/edx-remote-config/stage/lms.yml#L875) `JWT_PAYLOAD_USER_ATTRIBUTE_MAPPING: {},`, so I have to override it in the tests.